File: C1.UWP.FlexGrid/CS/FlexGrid101/Data/Song.cs
using System;
using System.IO;
using System.Reflection;
using System.Collections.Generic;
using System.Xml.Serialization;
using Windows.Storage;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;

namespace FlexGrid101
{
    public class Song
    {
        [Display(Name = "Name")]
        public string Name { get; set; }

        [Display(Name = "Album")]
        public string Album { get; set; }

        [Display(Name = "Artist")]
        public string Artist { get; set; }

        [Display(Name = "Duration")]
        public long Duration { get; set; }  // in milliseconds

        [Display(Name = "Size")]
        public long Size { get; set; }      // in bytes

        [Display(Name = "Rating")]
        public int Rating { get; set; }     // from 0 to 5
    }

    public class MediaLibraryStorage
    {
        public static async Task<List<Song>> Load()
        {
            try
            {
                Uri resourceUri;
                if (typeof(MediaLibraryStorage).GetTypeInfo().Module.Name.EndsWith("exe") || 
                    Windows.ApplicationModel.DesignMode.DesignModeEnabled)
                {
                    resourceUri = new Uri("ms-appx:///Resources/songs.xml");
                }
                else
                {
                    resourceUri = new Uri("ms-appx:///FlexGrid101Lib/Resources/songs.xml");
                }
                var file = await StorageFile.GetFileFromApplicationUriAsync(resourceUri);
                var fileStream = await file.OpenAsync(FileAccessMode.Read);
                var xmls = new XmlSerializer(typeof(List<Song>));
                return (List<Song>)xmls.Deserialize(fileStream.AsStream());
            }
            catch
            {
                throw new Exception(Strings.FileNotFoundException);
            }
        }
    }
}

File: C1.UWP.FlexGrid/VB/FlexGridSamples/Data/Song.vb
Imports System.ComponentModel.DataAnnotations
Imports System.Threading.Tasks
Imports Windows.Storage
Imports System.Xml.Serialization
Imports System.Collections.Generic
Imports System.Reflection
Imports System.IO
Imports System
Imports Windows.Storage.Streams

Public Class Song
    <Display(Name:="Name")>
    Public Property Name() As String

    <Display(Name:="Album")>
    Public Property Album() As String

    <Display(Name:="Artist")>
    Public Property Artist() As String

    <Display(Name:="Duration")>
    Public Property Duration() As Long
    ' in milliseconds
    <Display(Name:="Size")>
    Public Property Size() As Long
    ' in bytes
    <Display(Name:="Rating")>
    Public Property Rating() As Integer
    ' from 0 to 5
End Class

Public Class MediaLibraryStorage
    Public Shared Async Function Load() As Task(Of List(Of Song))
        Try
            Dim resourceUri As Uri
            If GetType(MediaLibraryStorage).GetTypeInfo().[Module].Name.EndsWith("exe") OrElse Windows.ApplicationModel.DesignMode.DesignModeEnabled Then
                resourceUri = New Uri("ms-appx:///Resources/songs.xml")
            Else
                resourceUri = New Uri("ms-appx:///FlexGridSamplesLib/Resources/songs.xml")
            End If
            Dim file As StorageFile = Await StorageFile.GetFileFromApplicationUriAsync(resourceUri)
            Dim fileStream As IRandomAccessStream = Await file.OpenAsync(FileAccessMode.Read)
            Dim xmls As New XmlSerializer(GetType(List(Of Song)))
            Return CType(xmls.Deserialize(fileStream.AsStream()), List(Of Song))
        Catch
            Throw New Exception(Strings.FileNotFoundException)
        End Try
    End Function
End Class